### ISSUE TITLE
Add shapes for ontology classes

### ISSUE THREAD

**Issue by cmungall** [geneontology/go-shapes#6]
(lower priority / question)

do we want to do things like
```
<BiologicalProcess> {
  a <BiologicalProcessClass> ;
...
}

<Cell> { 
  a <CellClass> ;
  ..
}

...

<CellClass> {
    // constraint URI to be CL or FAO or PO
}
```

This is maybe a bit circular with the tagger but seems like it would be useful to be able to be explicit about what kinds of classes are expected as the targets of rdf:type

**Comment by goodb** [comment]
Tangentially related question here.  I'm wondering if it would be better idea to go ahead and tag all the terms in the tbox accessible for go-cams with their top-level BL types in advance (in their pipeline build) rather than needing to apply the reasoner to each individual go-cam before we can use a shape.  It seems this would be broadly useful (e.g. it would solve another of Tremayne's problems with the UI code) and I'm worried that the reasoner step is currently (in my hands) rather slow.

**Comment by goodb** [comment]
Noting here that, after some further testing, the application of the reasoner for on-demand model tagging is not any slower than running the reasoner to check consistency and infer types as we have been doing in Noctua for a long time.  It literally only adds a few miliseconds to use it to add the types.  

Note related issue: https://github.com/geneontology/minerva/issues/231

**Comment by balhoff** [comment]
We've been using this pattern for a while now.
